tests/test_imputation/test_imputation.py

```diff
@@ -944,3 +946,152 @@ def test_backwards_imputation(self):
         )
 
         assert_frame_equal(df_result, expout_df)
+
+
+class TestRunImputation:
+    """Unit test for run_imputation"""
+
+    def input_data_run_imputation(self):
+        """Create input data for the run_imputation function"""
+        # Currently input_df isn't being used as
+        # fake data is hard coded into
+        # function until ingest is firmed down
+
+        # columns for the dataframe
+        input_cols = [
+            "reference",
+            "survey",
+            "checkletter",
+            "employees",
+            "202012_201",
+            "202012_202",
+            "202009_201",
+            "202009_202",
+        ]
+
+        # data in the column order above
+        input_data = [
+            [1, "2", "A", 100, 1, 1, 1, 1],
+            [2, "2", "A", 100, 11, 1, 10, 1],
+            [3, "2", "A", 100, 11, 1, 10, 1],
+            [4, "2", "A", 100, 11, 1, 10, 1],
+            [5, "2", "A", 100, 11, 1, 10, 1],
+            [6, "2", "A", 100, 11, 1, 10, 1],
+            [7, "2", "A", 100, 11, 1, 10, 1],
+            [8, "2", "A", 100, 11, 1, 10, 1],
+            [9, "2", "A", 100, 11, 1, 10, 1],
+            [10, "2", "A", 100, 11, 1, 10, 1],
+            [11, "2", "A", 100, 110, 1, 100, 1],
+            [12, "2", "A", 100, "missing", 1, 10, 1],
```

Comment:
Could we consider a different way of doing "missing" so it's not a string? For example what if we do the following:

`[12, "2", "A", 100, , 1, 10, 1],`

Reply:
I've now implemented this adding np.nan instead of "missing" as the column is integer type.